packages/yew/src/platform/rt_tokio/local_worker.rs

```diff
@@ -0,0 +1,190 @@
+                            while let Some(m) = rx.next().await {
+                                m();
+                            }
+                        });
+                    });
+                })?;
+        }
+
+        Ok(Self { task_count, tx })
+    }
+
+    pub fn task_count(&self) -> usize {
+        self.task_count.load(Ordering::Acquire)
+    }
+
+    pub fn spawn_pinned<F, Fut>(&self, f: F)
+    where
+        F: 'static + Send + FnOnce() -> Fut,
+        Fut: 'static + Future<Output = ()>,
+    {
+        let guard = LocalJobCountGuard::new(self.task_count.clone());
+
+        // We ignore the result upon a failure, this can never happen unless the runtime is
+        // exiting which all instances of Runtime will be dropped at that time and hence cannot
+        // spawn pinned tasks.
+        let _ = self.tx.unbounded_send(Box::new(move || {
+            spawn_local(async move {
+                let _guard = guard;
+
+                f().await;
+            });
+        }));
+    }
+}
+
+pub struct LocalJobCountGuard(Arc<AtomicUsize>);
+
+impl LocalJobCountGuard {
+    fn new(inner: Arc<AtomicUsize>) -> Self {
+        inner.fetch_add(1, Ordering::AcqRel);
+        LocalJobCountGuard(inner)
+    }
+}
+
+impl Drop for LocalJobCountGuard {
+    fn drop(&mut self) {
+        self.0.fetch_sub(1, Ordering::AcqRel);
+    }
+}
+
+#[derive(Debug, Clone)]
+pub(crate) struct LocalHandle {
+    // This type is not send or sync.
+    _marker: PhantomData<*const ()>,
+    task_count: Arc<AtomicUsize>,
+}
+
+impl LocalHandle {
+    pub fn try_current() -> Option<Self> {
+        // We cache the handle to prevent borrowing RefCell.
+        thread_local! {
+            static LOCAL_HANDLE: Option<LocalHandle> = TASK_COUNT
+            .with(|m| m.borrow().clone())
+            .map(|task_count| LocalHandle { task_count, _marker: PhantomData });
+        }
+
+        LOCAL_HANDLE.with(|m| m.clone())
+    }
+
+    pub fn current() -> Self {
+        Self::try_current().expect("outside of Yew runtime.")
+    }
+
+    pub fn spawn_local<F>(&self, f: F)
+    where
+        F: Future<Output = ()> + 'static,
+    {
+        let guard = LocalJobCountGuard::new(self.task_count.clone());
+
+        LOCAL_SET.with(move |local_set| {
+            local_set.spawn_local(async move {
+                let _guard = guard;
+
+                f.await;
+            })
+        });
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use futures::channel::oneshot;
+    use tokio::test;
+    use yew::platform::Runtime;
+
+    use super::*;
+
+    #[test]
+    async fn test_local_handle_exists() {
+        assert!(LocalHandle::try_current().is_none());
+
+        let runtime = Runtime::default();
+        let (tx, rx) = oneshot::channel();
+
+        runtime.spawn_pinned(move || async move {
+            tx.send(LocalHandle::try_current().is_some())
+                .expect("failed to send");
+        });
+
+        assert!(rx.await.expect("failed to receive"));
```

Comment:
I think it's a good idea to add a timeout here. If the runtime is never spawned, this test will just hang and never fail